overlord/configstate/config/transaction.go

```diff
@@ -361,3 +483,35 @@ func (e *NoOptionError) Error() string {
 	}
 	return fmt.Sprintf("snap %q has no %q configuration option", e.SnapName, e.Key)
 }
+
+// VirtualCfgFunc can be used for virtual "transaction.Get()" calls
+type VirtualCfgFunc func(snapName, key string) (result interface{}, err error)
```

Comment:
it's not entirely clear to me why this takes a snapName given that is going be registered under one?  to share the function across different snaps? but we don't have a concrete use case for that

Reply:
Indeed, YAGNI, I am killing it.